src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useState } from "react";
+import { addComment, getUsers } from "../api-utils";
+
+function CommentAdder({ article_id, setComments }) {
+  const [noUser, setNoUser] = useState(false);
+  const [users, setUsers] = useState([]);
+  const [error, setError] = useState(false);
+  const [inputs, setInputs] = useState({
+    username: "",
+    comment: "",
+  });
+  const commentLength = inputs.comment.length;
+
+  useEffect(() => {
+    getUsers().then((data) => {
+      setUsers(data);
+    });
+  }, []);
+
+  const handleChange = (event) => {
+    const value = event.target.value;
+    setInputs({ ...inputs, [event.target.name]: value });
+  };
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    setError(false);
+    users.forEach((user) => {
+      if (user.username !== inputs.username) {
+        return setNoUser(true);
+      }
+    });
+
+    addComment(inputs.username, inputs.comment, article_id)
+      .then((postedComment) => {
+        setComments((currentComments) => {
+          return [postedComment, ...currentComments];
+        });
+        setInputs({ username: "", comment: "" });
+        setNoUser(false);
+        alert("Your comment has been posted.");
+      })
```

Comment:
While this is comment is being posted as a user I can still click the submit button which will lead to multiple of the same comment being posted. Can you think of a way to prevent the user from being able to do this?